doc/content/integrations/cloud-integrations/blockbax/_index.md

```diff
@@ -0,0 +1,164 @@
+
+Check the [Blockbax documentation page](https://blockbax.com/docs/) for more info and the screenshot below for an impression.
+
+{{< figure src="subject-composition.gif" alt="Subject composition" >}}
+
+## Prerequisites
+
+1. Own a Blockbax project or [request one](https://blockbax.com/about#contact) if you do not have an account yet.
+
+## Blockbax Setup
+
+[Login](https://login.blockbax.com/) to the Blockbax Platform.
+
+Create a [subject type](https://blockbax.com/docs/subjects/#managing-subject-types) and configure metrics. External IDs for these metrics  need to match the property names returned by the uplink payload formatter that will be [configured on {{% tts %}}]({{< ref "/integrations/cloud-integrations/blockbax#configure-the-things-stack" >}}).
+
+{{< figure src="blockbax-metric-external-id-in-payload.png" alt="Relating Blockbax metrics to the Things Stack" >}}
+
+Additionally, you can choose to [create subjects](https://blockbax.com/docs/subjects/#creating-subjects) manually with external IDs matching the device IDs or set the `autoCreateSubjects` option in {{% tts %}} payload formatter function to `true`. In this case your subject will be created automatically if possible and there is no subject yet.
+
+## Configure {{% tts %}}
+
+Make sure your device is added to {{% tts %}}. See [Adding Devices](https://www.thethingsindustries.com/docs/devices/adding-devices/) for more info.
+
+On the left menu of your {{% tts %}} application, select **Payload formatters &#8594; Uplink**.
+
+Select **Javascript** payload formatter type.
+
+Paste the following contents in the **Formatter parameter** window:
+
+```javascript
+function decodeUplink(input) {
+    decoded = decoder(input.bytes)
+    return {
+        data: {
+            options: {
+                // Set this option to true to create subjects automatically
+                autoCreateSubjects: false
+            },
+
+            // For numeric metrics
+            <Blockbax-external-metric-ID>: decoded.exampleNumber,
+
+            // For text metrics
+            <Blockbax-external-metric-ID>: decoded.exampleText,
+
+            // For location metrics
+            <Blockbax-external-metric-ID>: {
+                lat: decoded.exampleLatitude,
+                lon: decoded.exampleLongitude
+            }
+        }
+    };
+}
+
+function decoder(bytes) {
+    // Add the logic that decodes the bytes from your device.
+    return decodedBytes
+}
+```
+
+The functions presented above need to be modified against your setup. See example below.
+
+<details><summary>Uplink payload formatter example</summary>
+
+```javascript
+    function decodeUplink(input) {
+      decoded = decoder(input.bytes)
+      return {
+        data: {
+          options: {
+          // Set this option to true to create subjects automatically
+            autoCreateSubjects: false
```

Comment:
I think this should be controlled via headers, and have a checkbox in the webhook template.

Reply:
@adriansmares Can you elaborate a bit more? You mean to create a custom Webhook Template? I wasn't aware this existed but just found it on your docs [here](https://www.thethingsindustries.com/docs/integrations/webhooks/webhook-templates/).

Reply:
> @adriansmares Can you elaborate a bit more? You mean to create a custom Webhook Template? I wasn't aware this existed but just found it on your docs [here](https://www.thethingsindustries.com/docs/integrations/webhooks/webhook-templates/).

Yes, you can follow the documentation that you linked to create a Blockbax Webhook template and simplify the process of creating the integration. You can also check our [webhook templates repository](https://github.com/TheThingsNetwork/lorawan-webhook-templates/) for the reference. Once the template is there, this PR would be good to go after slight changes.

Reply:
Thanks will look into it! Currently the platform does not support passing the `autoCreateSubjects` property so hope it is not a blocker to keep it in the payload like this (we will put it on our backlog to update this in the platform). All the other setup we can make much smoother via a webhook template, so will do that for sure!

Reply:
Most users will use payload formatters from the device repository, so they won't be able to fill this property in the payload formatter, so my suggestion is to look into loading it from other sources (headers, request parameters).

Reply:
I understand, but currently we only support passing it via the payload. I will discuss with the team if we can make it a todo to update this on our end. Hope you understand it will take some time and we can merge it for now and come back to improve on it later. 

For the other part we updated the docs and include a webhook template, thanks for the pointer!

Here is the PR for the webhook template: https://github.com/TheThingsNetwork/lorawan-webhook-templates/pull/50